torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -2731,6 +2731,16 @@ class TestPickler:
             torch.distributed.rpc.api._default_pickler is _internal_rpc_pickler
         )
 
+    @dist_init
+    def test_wait_all(self):
+        with rpc.wait_all():
+            self.assertTrue(rpc.thread_local_var.future_list == [])
+            dst = worker_name((self.rank + 1) % self.world_size)
+            fut = rpc.rpc_async(dst, torch.add, (torch.ones(2, 2), 1))
+            self.assertTrue(len(rpc.thread_local_var.future_list) == 1)
+            self.assertEqual(fut.wait(), torch.ones(2, 2) + 1)
```

Comment:
@mrshenli  I'm confused why we need this, seems independent of newly created function we are testing here.  It does however seem reasonable to, after the with block is over, test whether the future in future_list was waited by the context manager.  Is there a way to do this?

Reply:
Yep, you can access `fut` outside of the with context, as Python var lifetime is not scoped by context managers.

Reply:
@mrshenli But then is
``` self.assertEqual(fut.wait(), torch.ones(2, 2) + 1)```
actually needed or may I delete it?

Reply:
Let's keep this, just to make sure `rpc_async` still returns the correct future object.